besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java

```diff
@@ -300,8 +301,10 @@ public BesuController build() {
             reorgLoggingThreshold,
             dataDirectory.toString());
 
+    final CachedMerkleTrieLoader cachedMerkleTrieLoader = new CachedMerkleTrieLoader(metricsSystem);
```

Comment:
this feels like a worldstateArchive concern that is leaking.  Non-blocking, but I think we could benefit from a builder and/or pass metricsSystem instead